OPS-0 Adding functionatlity to get vpc facts and securit group facts

runderwoodcr14 opened the pull request:
# Description

Adding the functionality to helm role to be able to lookup for vpc id, which will be used then to get the security groups created with the role [ansible-role-aws-ec2-security-group](https://github.com/Flaconi/ansible-role-aws-ec2-security-group) based on the vpc id and the helm chart name.

Commit:
OPS-0 Adding functionatlity to get vpc facts and securit group facts

Commit:
OPS-0 Update gather facts for sg file name

Commit:
OPS-0 Updating role with changes to get SG facts

Commit:
OPS-0 Updating identation

Commit:
OPS-0 Update with line size fixing

Commit:
OPS-0 Added cluster_env variable

- Added cluster_env to the defaults with the default value of playground, this value its also set in hosts_vars > internet > all.yml

Commit:
OPS-0 Updated gather_facts_security_groups

- Using right variable to loop over on the task `Set EC2 Security Groups filter`
- Updated identation

Commit:
OPS-0 Updated run

- Set helm_charts_updated to an empty list
- Refactor how the security groups are assigned to the charts structure
- Added a new tadsk to render the charts with the security groups

Commit:
OPS-0 Update identenation for linter

Commit:
OPS-0 replace cluster_name with inventory_hostname

Commit:
OPS-0 Adding default region variable

Commit:
OPS-0 Update README

Commit:
OPS-0 Adding gitignore on the root of the project

runderwoodcr14 commented:
This is failing because the tests needs boto3 and botocore, if those modules are added then the test still will fail because it needs credentials to get vpc-facts and sg-facts:

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NoCredentialsError: Unable to locate credentials
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Unable to describe VPCs []: Unable to locate credentials"}
```
I tried this tests locally.
I have also run tests directly against playground and they work

danvaida commented on tasks/gather_facts_security_groups.yml:
```diff
@@ -0,0 +1,28 @@
+- name: Return EC2 Security Groups IDs
+  set_fact:
+    helm_ec2_security_groups: "{{ helm_ec2_security_groups |
+    combine( {item.item['tag:Helm_Chart']: {'Name': item.item['tag:Helm_Chart'], 'Groups': item |
```
I see what you did there 👍

danvaida reviewed (dismissed):
Nicely done 👍 
One thing though: regarding the credentials, you can use the CI/CD IAM technical user and have the tests run successfully.

Commit:
OPS-0 Workaround to avoid tests

- This is a workaround to avoid the tests to invoke gather facts of vpc and sg, this is not a fix, its just a way to get the tests passed while this gets fixed in a proper way

Commit:
OPS-0 Linter fix

- Linter :rage:
